compiler/src/dmd/safe.d

```diff
@@ -238,12 +238,20 @@ bool isSafeCast(Expression e, Type tfrom, Type tto)
  */
 bool checkUnsafeDotExp(Scope* sc, Expression e, Identifier id, int flag)
 {
-    if (!(flag & DotExpFlag.noDeref)) // this use is attempting a dereference
+    if (flag & DotExpFlag.noDeref) // this use is not attempting a dereference
+        return false;
+    if (id != Id.ptr)
+        return sc.setUnsafe(false, e.loc, "`(%s).%s` cannot be used in `@safe` code", e, id);
+
+    if (e.isStringExp()) // always null terminated
```

Comment:
does this only get string literals? it looks like any string expressions which is not what we want.

Reply:
String expressions are string literals.